compiler/tests/generated/constructor_parameters_types.cs

```diff
@@ -0,0 +1,158 @@
+
+
+// suppress "Missing XML comment for publicly visible type or member"
+#pragma warning disable 1591
+
+
+#region ReSharper warnings
+// ReSharper disable PartialTypeWithSinglePart
+// ReSharper disable RedundantNameQualifier
+// ReSharper disable InconsistentNaming
+// ReSharper disable CheckNamespace
+// ReSharper disable UnusedParameter.Local
+// ReSharper disable RedundantUsingDirective
+#endregion
+
+namespace Test
+{
+    using System.Collections.Generic;
+
+    [System.CodeDom.Compiler.GeneratedCode("gbc", "0.11.0.0")]
+    public enum TestEnum
+    {
+        EnumVal1,
+        EnumVal2,
+        EnumVal3,
+    }
+
+    [global::Bond.Schema]
+    [System.CodeDom.Compiler.GeneratedCode("gbc", "0.11.0.0")]
+    public partial class Simple
+    {
+        [global::Bond.Id(0)]
+        public int someInt { get; private set; }
+
+        [global::Bond.Id(1)]
+        public int anotherInt { get; private set; }
+
+        public Simple(
+            int someInt,
+            int anotherInt
+        )
+        {
+            this.someInt = someInt;
+            this.anotherInt = anotherInt;
+        }
+
+        public Simple()
+        {
+            
+        }
+    }
+
+    [global::Bond.Schema]
+    [System.CodeDom.Compiler.GeneratedCode("gbc", "0.11.0.0")]
+    public partial class Foo
+    {
+        [global::Bond.Id(0), global::Bond.Type(typeof(global::Bond.Tag.wstring))]
+        public string someText { get; private set; }
+
+        public Foo(
+            string someText
+        )
+        {
+            this.someText = someText;
+        }
+
+        public Foo()
+        {
+            someText = "BaseText1";
+        }
+    }
+
+    [global::Bond.Schema]
+    [System.CodeDom.Compiler.GeneratedCode("gbc", "0.11.0.0")]
+    public partial class Bar
+        : Foo
```

Comment:
Could you add a test with a class that derived from `Bar`?

Reply:
Added.  Thanks for pointing out that my code was not working for more than one level of derivation.